### ISSUE TITLE
Basic setup with sane options

### ISSUE THREAD

**Issue by itsz-san** [itsz-san/itsz-nvim#1]
## Detailed Description

Establish the directory structure for config files and make an initial configuration with some sane default options. The setup for the directory structure should reflect some understanding of how neovim loads configuration files. Additionally, it should support switching between different user configurations. **Configurations should be using Lua.**

## Context

In the past, when setting up configurations I tend to just cobble things together. Often, this leaves things in a state of mess as I go back and forth making changes without looking too deep into the technical details. Additionally, I tend to experiment with configurations shared by others. Being able to keep configuration files organized hopefully helps with maintainability. 

Additionally, as this is my virgin voyage, exploring neovim options and figuring out a set of initial defaults would seem like a safe way to launch myself into this adventure.

## Possible Implementation

Assume that this repository will be cloned into `~/.config/nvim`. The directory structure could be;

``` bash
~/.config/nvim/
├── lua/
│   ├── itsz/
│   │   ├── options.lua
│   │   └── init.lua # this loads my set of preferred configurations
│   └── another-user/
│       ├── options.lua
│       └── init.lua # this loads another set of configurations
└── init.lua # this should be clean(?) and source the preferred user configs
```

The above directory structure would allow me to switch between different configurations and store interesting configurations that are being shared in the wild. 

## Notes, Open Questions and Additional Information
**ADD-INFO**: We can tell Neovim to source from a different `init.lua` file at startup using the `-u /filepath/to/another/init.lua`. So I'm not sure if the proposed implementation is ideal. Different approach would matter if it affects performance in some way, but that's beyond the scope of this first baby step into building my own custom configuration setup.

**QUESTION**: How does Neovim load configuration files?

**QUESTION**: How might we measure performance? Not urgent now, but I reckon we might need a way to measure some kind of performance test when we extend the hyperextensible Vim-like text editor.

**NOTE**: In trying to draw the directory structure used above, I ended up using [this online tree-like utility for generating ASCII folder structure diagrams](https://tree.nathanfriend.io/).

**Comment by itsz-san** [comment]
After some deliberation, I think I'd want to keep it simple.

```
~/.config/nvim/
├── lua/
│   └── itsz/
│       ├── core/
│       │   ├── options.lua
│       │   └── keymaps.lua
│       └── init.lua
└── init.lua
```

There is no point in over-thinking how to create a modular configuration setup when the only category of settings I can think of is keymaps. At the very least, I want to separate configurations for key mappings into it's own file so that I can quickly access it to make changes.

**NOTE**: Official help page for options https://neovim.io/doc/user/options.html
**NOTE**: When writing out our sane options, include the [description](https://neovim.io/doc/user/quickref.html#option-list) for each option as trailing comments
```
vim.opt.cursorline = true -- highlight the screen line of the cursor
vim.opt.cursorcolumn = true -- highlight the screen column of the cursor
```

**Comment by itsz-san** [comment]
With regards to deciding what default options to write, perhaps we can start with looking at [NeoVim's defaults](https://neovim.io/doc/user/vim_diff.html).

**Comment by itsz-san** [comment]
## Writing Options
## TLDR; use `vim.opt`
`vim.opt` behaves like `:set`
`vim.opt_global` behaves like `:setglobal`
`vim.opt_local` behaves like `:setlocal`

In all honesty, I have no clue when it's appropriate to use them. I suppose that for a user defaults we would want to set the options globally. But that begs the question - **What's the difference between `:set` and `:setglobal`**?.

For now, I'm just going to use `vim.opt`. My current understanding is that, `vim.opt` will set the both global and local options, which is essentially what I want. But knowing all these might help later down the road when I try to implement different configurations for different file types. _Or something like that._

> Options are global when you mostly use one value for all buffers and windows.
For some global options it's useful to sometimes have a different local value.
You can set the local value with ":setlocal".  That buffer or window will then
use the local value, while other buffers and windows continue using the global
value.
>
> From https://neovim.io/doc/user/options.html

## Additional Info
Here's more stuffs to confuse myself. 

Global options:
`vim.api.nvim_set_option()`/`vim.api.nvim_get_option()`
or simply `vim.go`

Buffer-local options:
`vim.api.nvim_buf_set_option()`/`vim.api.nvim_buf_get_option()`
or simply `vim.bo`

Window-local options:
`vim.api.nvim_win_set_option()`/`vim.api.nvim_win_get_option()`
or simply `vim.wo`


## References
https://github.com/nanotee/nvim-lua-guide
https://www.reddit.com/r/neovim/comments/nn0b8f/comment/gzrtwa1/
https://neovim.io/doc/user/options.html#global-local
https://neovim.io/doc/user/lua.html#lua-vim-options

**Comment by itsz-san** [comment]
Geez, I'm a fool. I've been accessing the options-list on the browser when I could literally just do it from within NeoVim. To do so, just run `:help` and navigate to the appropriate section. ~~As of now, I'm not sure yet if I can jump straight into a particular section of the reference.~~

Also, I found out that we can change the options by running `:options`. The options are organised into categories, and perhaps I should organise my `options.lua` following the same categorisation.

PS: As it turns out, I can get the description of each options via `:help option-list`
